### ISSUE TITLE
toEqual throws Range Error when objects with circular references are compared

### ISSUE THREAD

**Issue by mayankchd** [mjackson/expect#50]
it('works when object has circular reference' , function () {
          function circular() {
            this.circularRef = this
          }
          var a = new circular()
          var b = new circular()
          expect(a).toEqual(b)
       })

    Throws RangeError: Maximum call stack size exceeded . 

As this package uses deep-equal to check for equality , similar issue has been filed on [deep-equal](https://github.com/substack/node-deep-equal/issues/19) 8 months ago .

I think using [deeper](https://www.npmjs.com/package/deeper) for checking for equality would be good as it works when objects with circular references are compared.
It passed all the tests along with circular reference one .

**Comment by mayankchd** [comment]
I can submit a PR + tests(for checking circular reference) for this.

**Comment by vvo** [comment]
@mayankchd I think it's a reasonable approache to use something that seems more maintained where all the bugs we need are fixed.

DO it!

**Comment by mjackson** [comment]
Released in v 1.13.1

**Comment by rstacruz** [comment]
Hm, maybe a bit of a regression here, I'm getting errors like these (extra line breaks mine):

```
Error: Expected
  { 'README.md': 'index.html', 'docs/install.md': 'install.html', 'docs/usage.md': 'usage.html' }
  to equal
  { 'README.md': 'index.html', 'docs/install.md': 'install.html', 'docs/usage.md': 'usage.html' }
```
